File: webapp/src/app/models/playlistViewDto.ts
import {SongViewDto} from "./songViewDto";

export interface PlaylistViewDto {
  id: number,
  name: string,
  type: string,
  createdDate: Date,
  updatedDate: Date,
  ownerUserId: number,
  ownerUserName: string,
  followableByUser: Boolean,
  unfollowableByUser: Boolean,
  songs: SongViewDto[]
}

File: musify/src/main/java/ro/ubb/postuniv/musify/dto/PlaylistViewDto.java
package ro.ubb.postuniv.musify.dto;

import java.sql.Date;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class PlaylistViewDto {

    private Integer id;
    private String name;
    private String type;
    private Date createdDate;
    private Date updatedDate;
    private Integer ownerUserId;
    private String ownerUserName;
    private List<SongViewDto> songs;
}
